docs/articles/documentation/guides/basic-guides/assert.md

```diff
@@ -138,6 +138,37 @@ To set the timeout when you launch tests, pass the timeout value to the [runner.
 method if you use API or specify the [assertion-timeout](../../reference/command-line-interface.md#--assertion-timeout-ms) option
 if you run TestCafe from the command line.
 
+### Smart Assertion Limitations
+
+Smart assertions auto retry feature only works with:
+
+* Promises returned from [`ClientFunctions`](../../reference/test-api/clientfunction/README.md),
+* [`Selector`](../../reference/test-api/selector/README.md) properties,
+* [`RequestLogger.count`](../../reference/test-api/requestlogger/count.md) properties,
+* [`RequestLogger.contains`](../../reference/test-api/requestlogger/contains.md) properties.
+
+The auto retry feature does not work with [DOM node shapshot](./select-page-elements.md#dom-node-snapshot) properties. If you execute the selector as an asynchronous function, its value is immediately resolved, doesn't update and the corresponding test fails:
+
+**Example:**
+
+```js
+test('Button click', async t => {
+    const btn = Selector('#btn');
+
+    await t
+        .click(btn)
+        .expect(await btn.textContent).contains('Loading...'); //fails because selector value doesn't update according to page behavior
+
+    await t
+        .click(btn)
+        .expect(btn()).contains({ innerText: 'Loading...'}); //fails because selector value doesn't update according to page behavior
+
+    await t
+        .click(btn)
+        .expect(btn.innerText).eql('Loading...'); //passes because promise returned from the selector eventually updates
+});
+```
```

Comment:
Can we also show the page code as an example:

```html
<html>
    <body>
        <div id="btn" onclick="window.setTimeout(() => this.innerText = 'Loading...', 5000)">Click me!</div>
    </body>
</html>
```

If I understand this right

Reply:
edited